### PR TITLE
Issue 511, implemented custom ConfigurationSection

### PR DESCRIPTION
**Pull request by tetrodoxin**
New PR. since previous PR failed due to file header unit test.
Replaced ConfigSectionHandler (since IConfigurationHandler is deprecated)  with XmlLoggingConfigurationSection.
No further major changes.
Adjusted respective unit test.

---
This change is [Reviewable](https://reviewable.io/reviews/nlog/nlog/890)

### CODE REVIEW

**Comment by 304NotModified** [comment]
Hey! 

A new PR is not needed (and not preferred) just commit and push to the branch you used for the PR. (next time)

File: src/NLog/Config/XmlLoggingConfigurationSection.cs [hunk 1]
```diff
@@ -31,29 +31,37 @@
...
     /// <summary>
-    /// NLog configuration section handler class for configuring NLog from App.config.
+    /// Class for providing Nlog configuration xml code from app.config
+    /// to <see cref="XmlLoggingConfiguration"/>
     /// </summary>
-    public sealed class ConfigSectionHandler : IConfigurationSectionHandler
+    public class XmlLoggingConfigurationSection : ConfigurationSection
     {
-        object Create(XmlNode section, IAppDomain appDomain)
+        private XmlLoggingConfiguration deserializedConfig = null;
```
Comment (304NotModified) [inline src/NLog/Config/XmlLoggingConfigurationSection.cs]: Don't rename please.  please keep it backwards compatible

**Comment by 304NotModified** [comment]
Closes #511

File: src/NLog/Config/XmlLoggingConfigurationSection.cs [hunk 2]
```diff
@@ -31,29 +31,37 @@
...
+    using System.Configuration;
+    using System.Xml;
 
     /// <summary>
-    /// NLog configuration section handler class for configuring NLog from App.config.
+    /// Class for providing Nlog configuration xml code from app.config
+    /// to <see cref="XmlLoggingConfiguration"/>
     /// </summary>
-    public sealed class ConfigSectionHandler : IConfigurationSectionHandler
+    public class XmlLoggingConfigurationSection : ConfigurationSection
```
Comment (304NotModified) [inline src/NLog/Config/XmlLoggingConfigurationSection.cs]: Don't rename please. please keep it backwards compatible

**Comment by 304NotModified** [comment]
Hi, I added one comment on your code. Can you change that in your PR?

**Comment by tetrodoxin** [comment]
I guess I've learned some things about contributing, for the next time.
Sorry it took that long, I hadn't anticipated my workload @work these days.

**Comment by 304NotModified** [comment]
No problem! 

Your pr looks good,  but we have to find out what the backwards compatibility effects on this change is,  before merging.

**Comment by tetrodoxin** [comment]
Something I could help with?

**Comment by 304NotModified** [comment]
We can do two things:

- Google about this change for  backwards compatibility
- Create a test app and test it with the latest NLog version (4.1.1) with a config in the XML, and then test it with a local build. (remove nlog references from the project and add the local build ones (.dll))

If you like to help with that, that would be great :)

**Comment by tetrodoxin** [comment]
Internet research didn't produce any hint for problems with backward compatibility, I'll check with ILSpy how GetSection() resolves internally and then write such an TestApp, but I'm affraid I can't test for Windows Phone.

**Comment by 304NotModified** [comment]
> but I'm affraid I can't test for Windows Phone.

Just .Net is good enough :)

**Comment by 304NotModified** [comment]
Hey @tetrodoxin ,

Did you had any change to test this as proposed?

**Comment by 304NotModified** [comment]
If you like to trigger Travis:

```
git commit -m "trigger Travis build" --allow-empty
git push
```

**Comment by tetrodoxin** [comment]
Hey Julian,

actually yes, finally. 
I created a small project, that just writes some silly sentences into different levels and loggers and additionally I took the liberty of testing it with one of our distributed projects at work :-)
But... how can I share the results? I don't really want to 'pollute' the nlog repo with that test project.

**Comment by 304NotModified** [comment]
Is an unit test project?

**Comment by tetrodoxin** [comment]
Yes, a very small solution, consisting of two test 'projects', including exactly the same source file (linked reference).
You may have a look at https://github.com/tetrodoxin/NLog_Legacy_Test

**Comment by 304NotModified** [comment]
@tetrodoxin 

Really thanks for this change. To be 100% sure, and because there is no functional requests for this, I will merge this into 5.0. And that could take a year, but we think it's really important to keep the breaking changes (binary breaking change here), as low as possible.

**Pull request merged**